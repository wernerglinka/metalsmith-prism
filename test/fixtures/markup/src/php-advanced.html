<code class="language-php">
&lt;?php
namespace App\Services;

use App\Interfaces\LoggerInterface;
use App\Traits\Loggable;

/**
 * Example service class demonstrating PHP features
 */
class AdvancedService implements LoggerInterface {
    use Loggable;

    private string $name;
    protected array $config = [];
    public readonly int $id;

    public function __construct(
        private LoggerInterface $logger,
        #[Required] string $name = 'default'
    ) {
        $this->name = $name;
    }

    public function process(mixed $data): array|null {
        try {
            $result = match($data['type']) {
                'string' => $this->processString($data['value']),
                'array' => $this->processArray($data['value']),
                default => throw new \InvalidArgumentException(),
            };

            return $result;
        } catch (\Throwable $e) {
            $this->logger->error($e->getMessage());
            return null;
        }
    }
}
</code>