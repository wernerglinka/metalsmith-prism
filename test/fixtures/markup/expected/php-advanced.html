<code class="language-php">
  <span class="token operator">&lt;</span>
  <span class="token operator">?</span>php <span class="token keyword">namespace</span>
  <span class="token package">App<span class="token punctuation">\</span>Services</span>
  <span class="token punctuation">;</span>
  <span class="token keyword">use</span>
  <span class="token package">App<span class="token punctuation">\</span>Interfaces<span class="token punctuation">\</span>LoggerInterface</span>
  <span class="token punctuation">;</span>
  <span class="token keyword">use</span>
  <span class="token package">App<span class="token punctuation">\</span>Traits<span class="token punctuation">\</span>Loggable</span>
  <span class="token punctuation">;</span>
  <span class="token comment">/** * Example service class demonstrating PHP features */</span>
  <span class="token keyword">class</span>
  <span class="token class-name-definition class-name">AdvancedService</span>
  <span class="token keyword">implements</span>
  <span class="token class-name">LoggerInterface</span>
  <span class="token punctuation">{</span>
  <span class="token keyword">use</span>
  <span class="token package">Loggable</span>
  <span class="token punctuation">;</span>
  <span class="token keyword">private</span>
  <span class="token keyword type-declaration">string</span>
  <span class="token variable">$name</span>
  <span class="token punctuation">;</span>
  <span class="token keyword">protected</span>
  <span class="token keyword type-declaration">array</span>
  <span class="token variable">$config</span>
  <span class="token operator">=</span>
  <span class="token punctuation">[</span>
  <span class="token punctuation">]</span>
  <span class="token punctuation">;</span>
  <span class="token keyword">public</span>
  <span class="token keyword">readonly</span>
  <span class="token keyword type-declaration">int</span>
  <span class="token variable">$id</span>
  <span class="token punctuation">;</span>
  <span class="token keyword">public</span>
  <span class="token keyword">function</span>
  <span class="token function-definition function">__construct</span>
  <span class="token punctuation">(</span>
  <span class="token keyword">private</span>
  <span class="token class-name type-declaration">LoggerInterface</span>
  <span class="token variable">$logger</span>
  <span class="token punctuation">,</span>
  <span class="token attribute">
    <span class="token delimiter punctuation">#[</span>
    <span class="token attribute-content">
      <span class="token attribute-class-name class-name">Required</span></span>
    <span class="token delimiter punctuation">]</span></span>
  <span class="token keyword type-declaration">string</span>
  <span class="token variable">$name</span>
  <span class="token operator">=</span>
  <span class="token string single-quoted-string">'default'</span>
  <span class="token punctuation">)</span>
  <span class="token punctuation">{</span>
  <span class="token variable">$this</span>
  <span class="token operator">-&gt;</span>
  <span class="token property">name</span>
  <span class="token operator">=</span>
  <span class="token variable">$name</span>
  <span class="token punctuation">;</span>
  <span class="token punctuation">}</span>
  <span class="token keyword">public</span>
  <span class="token keyword">function</span>
  <span class="token function-definition function">process</span>
  <span class="token punctuation">(</span>
  <span class="token keyword type-hint">mixed</span>
  <span class="token variable">$data</span>
  <span class="token punctuation">)</span>
  <span class="token punctuation">:</span>
  <span class="token keyword return-type">array</span>
  <span class="token operator">|</span>
  <span class="token keyword type-declaration">null</span>
  <span class="token punctuation">{</span>
  <span class="token keyword">try</span>
  <span class="token punctuation">{</span>
  <span class="token variable">$result</span>
  <span class="token operator">=</span>
  <span class="token keyword">match</span>
  <span class="token punctuation">(</span>
  <span class="token variable">$data</span>
  <span class="token punctuation">[</span>
  <span class="token string single-quoted-string">'type'</span>
  <span class="token punctuation">]</span>
  <span class="token punctuation">)</span>
  <span class="token punctuation">{</span>
  <span class="token string single-quoted-string">'string'</span>
  <span class="token operator">=&gt;</span>
  <span class="token variable">$this</span>
  <span class="token operator">-&gt;</span>
  <span class="token function">processString</span>
  <span class="token punctuation">(</span>
  <span class="token variable">$data</span>
  <span class="token punctuation">[</span>
  <span class="token string single-quoted-string">'value'</span>
  <span class="token punctuation">]</span>
  <span class="token punctuation">)</span>
  <span class="token punctuation">,</span>
  <span class="token string single-quoted-string">'array'</span>
  <span class="token operator">=&gt;</span>
  <span class="token variable">$this</span>
  <span class="token operator">-&gt;</span>
  <span class="token function">processArray</span>
  <span class="token punctuation">(</span>
  <span class="token variable">$data</span>
  <span class="token punctuation">[</span>
  <span class="token string single-quoted-string">'value'</span>
  <span class="token punctuation">]</span>
  <span class="token punctuation">)</span>
  <span class="token punctuation">,</span>
  <span class="token keyword">default</span>
  <span class="token operator">=&gt;</span>
  <span class="token keyword">throw</span>
  <span class="token keyword">new</span>
  <span class="token class-name class-name-fully-qualified">
    <span class="token punctuation">\</span>InvalidArgumentException</span>
  <span class="token punctuation">(</span>
  <span class="token punctuation">)</span>
  <span class="token punctuation">,</span>
  <span class="token punctuation">}</span>
  <span class="token punctuation">;</span>
  <span class="token keyword">return</span>
  <span class="token variable">$result</span>
  <span class="token punctuation">;</span>
  <span class="token punctuation">}</span>
  <span class="token keyword">catch</span>
  <span class="token punctuation">(</span>
  <span class="token class-name class-name-fully-qualified">
    <span class="token punctuation">\</span>Throwable</span>
  <span class="token variable">$e</span>
  <span class="token punctuation">)</span>
  <span class="token punctuation">{</span>
  <span class="token variable">$this</span>
  <span class="token operator">-&gt;</span>
  <span class="token property">logger</span>
  <span class="token operator">-&gt;</span>
  <span class="token function">error</span>
  <span class="token punctuation">(</span>
  <span class="token variable">$e</span>
  <span class="token operator">-&gt;</span>
  <span class="token function">getMessage</span>
  <span class="token punctuation">(</span>
  <span class="token punctuation">)</span>
  <span class="token punctuation">)</span>
  <span class="token punctuation">;</span>
  <span class="token keyword">return</span>
  <span class="token constant">null</span>
  <span class="token punctuation">;</span>
  <span class="token punctuation">}</span>
  <span class="token punctuation">}</span>
  <span class="token punctuation">}</span></code>